### PR TITLE
8 comments

### PR DESCRIPTION
**Pull request by olivershh**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -70,11 +69,71 @@ describe("/api/articles", () => {
...
+      return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then((response) => {
+          const { body } = response;
+          const noCommentArticles = [4, 10, 2, 11, 12, 7, 8];
+          body.articles.forEach((article) => {
+            if (noCommentArticles.includes(article.article_id)) {
+              expect(article.comment_count).toBe(0);
+              // working here
```
Comment (popatre) [inline __tests__/app.test.js:82]: remove comment

File: __tests__/app.test.js [hunk 2]
```diff
@@ -70,11 +69,71 @@ describe("/api/articles", () => {
...
+          const { body } = response;
+          expect(body.articles).toBeSortedBy("created_at", {
+            descending: true,
+          });
+        });
+    });
+    test("200: articles are filtered by topic, if provided", () => {
+      return request(app)
+        .get("/api/articles?topic=mitch")
+        .expect(200)
```
Comment (popatre) [inline __tests__/app.test.js:101]: I'd check the length to make sure you're not just getting back them all, and are only getting the amount related to mitch

File: models/articles.model.js [hunk 3]
```diff
@@ -27,15 +29,39 @@ exports.updateArticleById = (articleId, votes) => {
...
-  GROUP BY articles.article_id;`
+      `SELECT articles.article_id, title, topic, articles.author, articles.created_at, articles.votes, count(comments.article_id) AS comment_count 
+      FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id ${topicFilter}
+      GROUP BY articles.article_id ORDER BY articles.created_at DESC;`
     )
     .then((queryData) => {
+      const articles = queryData.rows;
+      articles.map(
+        (article) => (article.comment_count = parseInt(article.comment_count))
+      );
```
Comment (popatre) [inline models/articles.model.js:48]: If you delete this, does it still work? 
You dont actually do anything with `articles` in this block

**Review by popatre — APPROVED** [review]
Just a slight tweak to a test please.
Also, a question whether you need a bit of your code

File: __tests__/app.test.js (hunk 1 above)
Comment (olivershh) [reply __tests__/app.test.js:82]: done

File: __tests__/app.test.js (hunk 2 above)
Comment (olivershh) [reply __tests__/app.test.js:101]: done thanks

**Pull request merged**